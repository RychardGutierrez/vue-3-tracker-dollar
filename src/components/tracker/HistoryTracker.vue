<template>
  <div class="max-w-md w-full py-12 text-dollar-blue" v-if="history">
    <h2 class="mb-4 text-center text-lg">7 Day Tracking history 1 USD</h2>
    <div v-for="(rate, key) in history.quotes" :key="key"
      class="flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2">
      <p class="flex-1">{{ dateWeekDay(key) }}</p>

      <template class="flex gap-2 flex-1 justify-between" v-for="(value, key, index) in rate" :key="index">
        <p>{{ value }} {{ simbolMoney }}</p>
      </template>
    </div>
  </div>
</template>

<script setup>
defineProps({
  history: {
    typeof: Array,
    require: true
  },
  dateWeekDay: {
    typeof: Function,
    require: false,
    default: () => { }
  },
  simbolMoney: {
    typeof: String,
    require: true
  }
})
</script>

<style lang="scss" scoped>

</style>