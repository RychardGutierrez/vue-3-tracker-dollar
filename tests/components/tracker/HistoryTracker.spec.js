import { shallowMount } from '@vue/test-utils';

import HistoryTracker from '../../../src/components/tracker/HistoryTracker.vue';

describe('HistoryTracker composable', () => {
  test('should to match with snapshot', () => {
    const props = {
      history: {
        quotes: {
          '2022-09-19': {
            USDBOB: 6.828416,
          },
          '2022-09-20': {
            USDBOB: 6.90908,
          },
          '2022-09-21': {
            USDBOB: 6.915233,
          },
          '2022-09-22': {
            USDBOB: 6.914142,
          },
          '2022-09-23': {
            USDBOB: 6.90469,
          },
          '2022-09-24': {
            USDBOB: 6.95361,
          },
          '2022-09-25': {
            USDBOB: 6.95361,
          },
        },
      },
      simbolMoney: 'BO',
      dateWeekDay: jest.fn(() => 'Monday'),
    };
    const wrapper = shallowMount(HistoryTracker, { props });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
