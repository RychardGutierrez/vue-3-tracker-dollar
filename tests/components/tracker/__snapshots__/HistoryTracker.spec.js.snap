// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryTracker composable should to match with snapshot 1`] = `
"<div class=\\"max-w-md w-full py-12 text-dollar-blue\\">
  <h2 class=\\"mb-4 text-center text-lg\\">7 Day Tracking history 1 USD</h2>
  <div class=\\"flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2\\">
    <p class=\\"flex-1\\">Monday</p>
    <p>6.828416 BO</p>
  </div>
  <div class=\\"flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2\\">
    <p class=\\"flex-1\\">Monday</p>
    <p>6.90908 BO</p>
  </div>
  <div class=\\"flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2\\">
    <p class=\\"flex-1\\">Monday</p>
    <p>6.915233 BO</p>
  </div>
  <div class=\\"flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2\\">
    <p class=\\"flex-1\\">Monday</p>
    <p>6.914142 BO</p>
  </div>
  <div class=\\"flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2\\">
    <p class=\\"flex-1\\">Monday</p>
    <p>6.90469 BO</p>
  </div>
  <div class=\\"flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2\\">
    <p class=\\"flex-1\\">Monday</p>
    <p>6.95361 BO</p>
  </div>
  <div class=\\"flex py-2 px-12 bg-dollar-secondary text-white rounded-md shadow-md mb-2\\">
    <p class=\\"flex-1\\">Monday</p>
    <p>6.95361 BO</p>
  </div>
</div>"
`;
